### PR TITLE
Add sticky-session to types

### PR DESCRIPTION
**Pull request by vanny96**
Please fill in this template.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
         "lint": "dtslint types"
     },
     "devDependencies": {
+        "@types/node": "^12.12.14",
```
Comment (amcasey) [inline package.json]: Why was this needed?

**Review by amcasey — CHANGES_REQUESTED** [review]

**Comment by vanny96** [comment]
@amcasey sorry for the mistake! It's my first time contributing so I made some mistakes, if there is another error please tell me!

File: package.json (hunk 1 above)
Comment (vanny96) [reply package.json]: It wasn't sorry! I updated the pull request, let me know if I have to update anything!

File: types/sticky-session/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * Method to create the sticky session
+ * @param server Server used to create the session
+ * @param port Port where the app will be listening at
+ * @param options Options to customize the sticky session
+ */
+export function listen(
+    server: Server,
+    port: number,
+    options: {
```
Comment (demensky) [inline types/sticky-session/index.d.ts]: `options` is [optional](https://github.com/indutny/sticky-session/blob/master/lib/sticky/api.js#L12).

File: types/sticky-session/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        workers: number
```
Comment (demensky) [inline types/sticky-session/index.d.ts]: `workers` is [optional](https://github.com/indutny/sticky-session/blob/master/lib/sticky/api.js#L15).

File: types/sticky-session/index.d.ts (hunk 2 above)
Comment (demensky) [inline types/sticky-session/index.d.ts]: Please add [`env`](https://github.com/indutny/sticky-session/blob/master/lib/sticky/api.js#L17).

File: types/sticky-session/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Method to create the sticky session
+ * @param server Server used to create the session
+ * @param port Port where the app will be listening at
+ * @param options Options to customize the sticky session
+ */
+export function listen(
+    server: Server,
+    port: number,
```
Comment (demensky) [inline types/sticky-session/index.d.ts]: `port` is [optional](https://github.com/indutny/sticky-session/blob/master/lib/sticky/api.js#L18) ([`net.Server#listen`](https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback)).

File: types/sticky-session/index.d.ts (hunk 3 above)
Comment (demensky) [inline types/sticky-session/index.d.ts]: I propose to add `readonly`.

**Review by demensky — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (amcasey) [reply package.json]: Thanks!

File: types/sticky-session/index.d.ts (hunk 2 above)
Comment (vanny96) [reply types/sticky-session/index.d.ts]: I added ```env``` with ```any``` type because this is what the ```cluster.fork(env)``` method uses. If there is any correction to do let me know!

Comment (demensky) [reply types/sticky-session/index.d.ts]: `any` here is probably better.

File: types/sticky-session/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// Type definitions for sticky-session 1.1
+// Project: https://github.com/indutny/sticky-session
+// Definitions by: Giovanni Barbaro <https://github.com/vanny96>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import { Server } from "http";
+import { ProcessEnvOptions } from "child_process";
```
Comment (demensky) [inline types/sticky-session/index.d.ts]:
```suggestion
```

You just need to remove the `ProcessEnvOptions` which is probably left by accident.

**Review by demensky — CHANGES_REQUESTED** [review]

**Review by amcasey — APPROVED** [review]
LGTM after making the last fix suggested by @demensky

**Review by demensky — APPROVED** [review]

**Review by demensky — CHANGES_REQUESTED** [review]
Ups... wrong button.

**Comment by vanny96** [comment]
Sorry for the delay! It was a busy week so I had no time to check back! Anyway I removed the unused import, let me know if there is anything more I could do!

File: types/sticky-session/index.d.ts (hunk 5 above)
Comment (vanny96) [reply types/sticky-session/index.d.ts]: Yes I left them when I was playing around with the `env` type, I then opted for `any` instead but forgot to remove the import! I have updated the pull request! Let me know if there is anything more I have to change!

**Review by demensky — APPROVED** [review]

**Comment by orta** [comment]
Alright, this looks good 👍
